luma/core/framebuffer.py

```diff
@@ -25,57 +27,71 @@ class diff_to_previous(object):
     :param device: The target device, used to determine the initial 'previous'
         image.
     :type device: luma.core.device.device
+    :param num_segments: The number of segments to partition the image into. This
+        generally must be a square number (1, 4, 9, 16, ...) and must be able to
+        segment the image entirely in both width and height. i.e setting to 9 will
+        subdivide the image into a 3x3 grid when comparing to the previous image.
+    :type num_segments: int
     """
-    def __init__(self, device):
-        self.image = Image.new(device.mode, device.size, "white")
-        self.bounding_box = None
 
-    def redraw_required(self, image):
+    def __init__(self, num_segments=4, debug=False):
+        self.__debug = debug
+        self.__n = int(math.sqrt(num_segments))
+        assert num_segments >= 1 and num_segments == self.__n ** 2
+        self.prev_image = None
+
+    def redraw(self, image):
         """
-        Calculates the difference from the previous image, return a boolean
-        indicating whether a redraw is required. A side effect is that
-        ``bounding_box`` and ``image`` attributes are updated accordingly, as is
-        priming :py:func:`getdata`.
+        Calculates the difference from the previous image, returning a sequence of
+        image sections and bounding boxes that changed since the previous image.
```

Comment:
this sentence probably needs another linebreak so the note is separated by a newline.

Reply:
Done